### ISSUE TITLE
Send saved Examples

### ISSUE THREAD

**Issue by cawoodm** [a85/POSTMan-Chrome-Extension#3694]
I noticed the new examples feature and decided to re-organize all my requests. So instead of having 5 requests for each object (GET, PUT, POST, DELETE and SEARCH) I made one request per object and created 5 examples.

However, there seems to be no way to "Send" an example request. That's a pity - could this be added?

![image](https://user-images.githubusercontent.com/498834/32160915-fdbf69c0-bd53-11e7-87f5-6b010b2eb87b.png)

**Comment by prashantagarwal** [comment]
@cawoodm We are already tracking this here #3630 . I am closing this as duplicate.

**Comment by cawoodm** [comment]
You're sure it's the same issue? The OP of 3630 did not mention the "Examples" feature.

**Comment by prashantagarwal** [comment]
@cawoodm Yes in that ticket Examples are referred as saved responses.

**Comment by neilgoldman** [comment]
I do not think these are the same issues, In 3630, OP can still send requests after pressing the back button, it sounds like a quality of life feature.

For me, when using the Examples feature, that request is basically gone. I have a tab with it, but I cannot send any requests and I cannot save the Example (I get "The example could not be saved since it's parent request has been deleted.") I have to copy and paste the content of one tab over to a new tab to send any requests or save it ever again.

**Comment by a85** [comment]
@neilgoldman I'll reopen this. Rather than having a "Send" button in examples, we'll look at ways to copy over the saved request from an example into the current tab. @vkaegis Let's review this flow once.

**Comment by richardjprice** [comment]
I've just hit this particular scenario. It really feels like a missing killer feature that will provide a nice way to validate your examples are correct and working.

**Comment by mackaaij** [comment]
Same here. I have an odd endpoint I'd like to document a few examples for. Instead of adding the same request url + params multiple times just to switch the body, I wanted to add some examples. But people cannot send the examples... If you'd follow @a85's suggestion and copy over the request, consider keeping the request as an example instead of just overwriting (or at least notify the user the base request will be overwritten and can be saved like that).

**Comment by Ian1971** [comment]
I don't get why this is still missing. Seems obvious. A weird work around is that I can get the code snippet and run it via something else, but not in postman.

**Comment by 4islam** [comment]
Is it still missing?

**Comment by simplenotezy** [comment]
I also find it a really weird design decision to remove the send button from the example snippet. Why, whyyy? 😄

**Comment by uzi0espil** [comment]
I also ran into a situation where I want to re-run my examples. It would be great to support this issue.

**Comment by Manas-Nagelia** [comment]
2021 and nothing so far. It's annoying copying the request over and over again. What I did is that I just created a new collection called, "Example Requests", where I keep these example requests, but this is very inconvienent.

**Comment by filiptrplan** [comment]
Still waiting on this. I would love to choose from multiple different bodies for my requests. Would help a lot with debugging.

**Comment by Manas-Nagelia** [comment]
> Still waiting on this. I would love to choose from multiple different bodies for my requests. Would help a lot with debugging.

Yeah. It should be relatively simple to implement this.

**Comment by nicolasiensen** [comment]
I find myself copying the body of the examples and pasting it in the body of the request so I can send it 😅

**Comment by Manas-Nagelia** [comment]
> I find myself copying the body of the examples and pasting it in the body of the request so I can send it 😅

Yeah, I do the same right now, but it's very inconvenient.

**Comment by vmickus** [comment]
And I thought that I was just dumb and couldn't find the button to re-run my examples! :sweat_smile: 

Considering how long this is open, I don't think we will have a solution for this. That`s disappointing.

**Comment by Manas-Nagelia** [comment]
Yeah, I agree.

**Comment by ExaltedA** [comment]
A feature which would make lives easier

**Comment by lucidlips** [comment]
+1 seems like an obvious missing feature.

**Comment by gustavorps** [comment]
+1 obvious missing feature.

**Comment by ahmadelmalah** [comment]
Yeah, it's a missing feature!

**Comment by lukalopusina** [comment]
+1, It would be really helpful to be able to run each example separatly without need to copy/paste example body into request body to test every case.

**Comment by MacKinley** [comment]
Any workflow decisions would be great too! Mostly whether the Postman team wants to implement the "Send" button on the examples or implement a "Copy onto current tab" feature. That way the community could potentially work on it.

**Comment by STeeL835** [comment]
Come on, this is almost a must-have. An additional button that would copy request from an example to a new tab related to the example's postman request. Postman already can handle different tabs for the same saved request, this would just open a new one. 
And it would help with organizing collections, because you wouldn't need to save the slight variations as separate requests.

**Comment by richetoh** [comment]
still waiting on this

**Comment by haile-kaligo** [comment]
Still waiting... 🙏 🙏 🙏

**Comment by shikhartiwari96** [comment]
what are we?
Developers
What do we want?
A simple send button on examples

**Comment by Ian1971** [comment]
when do we want it?
Oct 30, 2017

**Comment by gustavorps** [comment]
![image](https://user-images.githubusercontent.com/516827/159950345-0082b7da-b8a7-4a28-9b8c-a9161d2dc077.png)

**Comment by sniper0** [comment]
+1

**Comment by muxfe** [comment]
+1

**Comment by chrisdreams13** [comment]
+1

**Comment by mike-loux-planview** [comment]
I'm a little surprised this has been open for so long - this seems like it would be low-hanging fruit.

**Comment by rcbyron** [comment]
🙏 🙏 🙏 🙏

**Comment by hunainkapadia** [comment]
much needed

**Comment by Seayon** [comment]
I recently used a tools similar to postman called apifox,and the have a "save run as test case" feature that works very well.

**Comment by DarkFisk** [comment]
Please, add send button

**Comment by nvuillam** [comment]
It seems that 5 years later I'm here for the same reason than all of you.... :)

**Comment by dben89x** [comment]
Good lord, how is this still not implemented? Thread was created back in 2017. This should have been in version 1.0.0. What's the reasoning for not having a send button on examples? I'm genuinely trying to think of any possible reason and can't come up with anything. I guess this is why us devs need project managers? Because common sense often escapes the best engineers apparently??

**Comment by Ian1971** [comment]
5 years is nothing. https://github.com/postmanlabs/postman-app-support/issues/285

**Comment by giridharvc7** [comment]
This might look a bit basic, but Can someone help us understand what you are really looking for by sending an example. 


Some context on the current system: 
An example has the following -> Request URL (with params), Headers and Body. What the Examples does not have are the following: Authorization, scripts apart from the response itself. 

This means that when you send a saved example (in its current form), you might miss out on some of the values that were used to generate the example itself. 

Asking the following because it'll help us understand your use case better and derive at a solution that solves for the use case.

**Comment by STeeL835** [comment]
@giridharvc7 
> it would help with organizing collections, because you wouldn't need to save the slight variations as separate requests.

Endpoints that are taking complex queries are pretty common, and my case is this: 

* I use collections generated from OpenAPI definition _file_
* I group requests by controllers and actions (endpoints) become Requests
* For complex endpoints I usually save different payloads as different examples rather than duplicate existing endpoints in the list
  * Testers use this feature to save responses, but when you want to recheck the request, you can't 
  * API developers can add examples to OpenAPI definitions, without the need to blow the collection up with unnecessary forks of the original Request, so that users can try them effortlessly while seeing a neat and organized collection and the variations are not shown like they are separate endpoints

As for additional data like auth and scripts - they exist in the Request these Examples attached to, so for my use case it's completely okay to inherit ones from the Request (especially when auth is usually inherited from parent and the scripts usually are compatible with the endpoint as a whole)

**Comment by shikhartiwari96** [comment]
> @giridharvc7
> 
> > it would help with organizing collections, because you wouldn't need to save the slight variations as separate requests.
> 
> Endpoints that are taking complex queries are pretty common, and my case is this:
> 
> * I use collections generated from OpenAPI definition _file_
> * I group requests by controllers and actions (endpoints) become Requests
> * For complex endpoints I usually save different payloads as different examples rather than duplicate existing endpoints in the list
>   
>   * Testers use this feature to save responses, but when you want to recheck the request, you can't
>   * API developers can add examples to OpenAPI definitions, without the need to blow the collection up with unnecessary forks of the original Request, so that users can try them effortlessly while seeing a neat and organized collection and the variations are not shown like they are separate endpoints
> 
> As for additional data like auth and scripts - they exist in the Request these Examples attached to, so for my use case it's completely okay to inherit ones from the Request (especially when auth is usually inherited from parent and the scripts usually are compatible with the endpoint as a whole)

+1

**Comment by ExaltedA** [comment]
> > @giridharvc7
> > > it would help with organizing collections, because you wouldn't need to save the slight variations as separate requests.
> > 
> > 
> > Endpoints that are taking complex queries are pretty common, and my case is this:
> > 
> > * I use collections generated from OpenAPI definition _file_
> > * I group requests by controllers and actions (endpoints) become Requests
> > * For complex endpoints I usually save different payloads as different examples rather than duplicate existing endpoints in the list
> >   
> >   * Testers use this feature to save responses, but when you want to recheck the request, you can't
> >   * API developers can add examples to OpenAPI definitions, without the need to blow the collection up with unnecessary forks of the original Request, so that users can try them effortlessly while seeing a neat and organized collection and the variations are not shown like they are separate endpoints
> > 
> > As for additional data like auth and scripts - they exist in the Request these Examples attached to, so for my use case it's completely okay to inherit ones from the Request (especially when auth is usually inherited from parent and the scripts usually are compatible with the endpoint as a whole)
> 
> +1

++

**Comment by Ian1971** [comment]
Anything not defined in the example it should inherit from the request it is attached to. So e.g. if the example has a header that is in the request as well, then the example header should be used

**Comment by DarkFisk** [comment]
> This might look a bit basic, but Can someone help us understand what you are really looking for by sending an example.

I was looking for functionality that allows me to send requests to the same endpoint with different parameters/body for example create user with different roles/rights. Currently, I have to clone the same request many times. In case something was changed on the backend I have to edit all of them since they don't have some common endpoint metadata, like headers/url/ect.

**Comment by giridharvc7** [comment]
@STeeL835 @DarkFisk thank you for the going through your use case! This really helps. 

Would love to see your workflow in action, just to understand things a bit more deeper, would you be open to having a call with me? This will help me understand your workflows in detail and also provide you with an opportunity to address anything else that's been bothering you. 

You can use [this link](https://calendly.com/giridhar-vc-postman/hello) to set up a call with me.

**Comment by presiden** [comment]
> This might look a bit basic, but Can someone help us understand what you are really looking for by sending an example.
> 
> Some context on the current system: An example has the following -> Request URL (with params), Headers and Body. What the Examples does not have are the following: Authorization, scripts apart from the response itself.
> 
> This means that when you send a saved example (in its current form), you might miss out on some of the values that were used to generate the example itself.
> 
> Asking the following because it'll help us understand your use case better and derive at a solution that solves for the use case.

I create an API for learning, with many fields of form-data include upload file. I hit the API several times for input data to database, i should save the example for each time hit API. Because for some cases (restructure table, some testing) , i deleted data in database and need to hit the API with same data.

Without send button, i must copy form-data by click bulk-edit, then create new request, paste form-data and retype the url.

**Comment by medin-n** [comment]
bump

**Comment by venriq** [comment]
If I were to write the story to implement this feature, it would be something like this:

**Why**
- As a software engineer responsible for designing Restful APIs in Postman
- I want to create variations **_(Examples)_** of a parent **Request**
- And be able to Send those requests
- So I don't have to duplicate the same **Request** over and over again just to make minimum changes

**Acceptance Criteria**

> **Scenario 1**
> - Given I have a **Workspace** in Postman
> - And that **Workspace** has a **Collection**
> - And that **Collection** has a **Request**
> - When I create an **Example** for that **Request**
> - Then the **Example** inherits the attributes from the parent **Request**
> - And when I modify specific attributes in the **Example**
> - Then, those modifications take precedence over the ones from the parent **Request**

> **Scenario 2**
> - Given I have **Request** with multiple examples
> - When I select any **Example**
> - Then I should see a **Send** button
> - And when I hit the **Send** button
> - Then, a **Request** should be sent based on the attributes defined in the selected **Example**

**Comment by ksenia-loseva** [comment]
almost 5 years passed... please add this feature! would save lots of time

**Comment by Sigtechnica** [comment]
pretty sure this thread is set to ignore :see_no_evil: 
otherwise I think we would have had a response already ...

**Comment by STeeL835** [comment]
> pretty sure this thread is set to ignore 🙈 otherwise I think we would have had a response already ...

We've had: https://github.com/postmanlabs/postman-app-support/issues/3694#issuecomment-1157496060

Unfortunately I didn't set up a call with them, because now I'm working in another place where we don't use Postman at all (at least yet - the openapi spec is generated by Swashbuckle and in a multiple services environment it's easier for us to just use swagger page than to write scripts that would regenerate postman collection file from downloaded swagger.json which we would need to import manually every time because postman can't be automated in this scenario)

**Comment by lorisantiagocm** [comment]
+1

**Comment by aslng** [comment]
It's sad that the feature hasn't been implemented after 5 years.
It's a feature that would be very useful for many!

Hope the feature will show up 'soon'. Thanks!

**Comment by remy-burney-pf** [comment]
+1

**Comment by BadiehTraboulsi** [comment]
Adding my voice to the rest!

**Comment by dantebarba** [comment]
haha today I asked myself 'why should I copy paste the query parameters from the examples it should be a better way of doing this obviously I'm quite incompetent' Glad to find this and see I wasn't the only one who felt incompetent.

+1

PS: I guess the best (and easier option) is to just add a "copy request" workflow. It should copy the request load into the parent

**Comment by Ian1971** [comment]
Welcome to the party

**Comment by chrisdreams13** [comment]
Could this post be nominated to the most wanted and ignored request?

**Comment by Ian1971** [comment]
> Could this post be nominated to the most wanted and ignored request?

Not even close. https://github.com/postmanlabs/postman-app-support/issues/285#issuecomment-1281779059

**Comment by donhuvy** [comment]
For temporary, you can use cURL: 
![image](https://user-images.githubusercontent.com/1328316/207497803-71fa5631-13d5-4ab4-8b3d-65411a258428.png)

Use Github bash (Linux shell) on Windows.

![image](https://user-images.githubusercontent.com/1328316/207498839-d8ea9804-04db-47f7-bb42-ce3a49be257e.png)

```bash
curl --location --request POST 'http://localhost:8082/api/auth/signin' \
--header 'Content-Type: application/json' \
--header 'Accept: */*' \
--data-raw '{
  "password": "123456a@",
  "username": "donhuvy@hotmail.com"
}'
```
I hope have button Send

**Comment by arlemi** [comment]
Hey everyone! We've added the possibility to send example requests from Postman v10.10, you'll now see a "Try it" button in your saved examples so you do not have to copy/paste its content to the request anymore. 

Make sure you've got the latest version of the app and give it a go! 

![example](https://user-images.githubusercontent.com/5029719/219416371-f3c462c6-6e83-4914-ac92-055c95aa80cf.gif)

**Comment by ExaltedA** [comment]
🥳

**Comment by dantebarba** [comment]
❤️ thanks!

**Comment by mike-loux-planview** [comment]
YASSSSSSSS

**Comment by shikhartiwari96** [comment]
Finally winter is over

**Comment by vcarballo** [comment]
Thanks, highly appreciated!! 🚀

**Comment by Ian1971** [comment]
omg

**Comment by giridharvc7** [comment]
We'd love to see how you folks use this, please use this issue or create a new issue for any specific feedback around this

**Comment by aslng** [comment]
I upgrade my postman on Apple Mac M1 machine. It now has postman version 10.10.3. However, I don't see the "Try" button after selecting an example. What may I miss?

**Comment by coditva** [comment]
@aslng Can you share a screen recording for the issue? You should be able to see a "Try" button for Examples saved for HTTP requests.

**Comment by aslng** [comment]
@coditva pls find screenshot below.
<img width="1507" alt="Screen Shot 2023-02-22 at 10 21 17 PM" src="https://user-images.githubusercontent.com/25257153/220834017-f626101c-fe10-46d9-a9be-7e42c9f3a235.png">

**Comment by giridharvc7** [comment]
@aslng 
Currently this feature is not available in scratchpad and only available for signed-in workspaces. We'll let you know once we pick this up for scratchpad

**Comment by hungtooc** [comment]
I save the examples as test cases, and others on the team will want to rerun that test case to check for errors. why do i have to copy the parameters one by one over to a new api to be able to run it?
This issue and [auto saved/save all](https://github.com/postmanlabs/postman-app-support/issues/8195#issue-584257737) are important features and depveloper needs, why don't you guys support them?

**Comment by giridharvc7** [comment]
@hungtooc Copying is not needed anymore, as mentioned here this feature has been released. Please update to the latest version of the app to see these changes. 

> Hey everyone! We've added the possibility to send example requests from Postman v10.10, you'll now see a "Try it" button in your saved examples so you do not have to copy/paste its content to the request anymore.
> 
> Make sure you've got the latest version of the app and give it a go!
> 
> ![example](https://user-images.githubusercontent.com/5029719/219416371-f3c462c6-6e83-4914-ac92-055c95aa80cf.gif) [ ![example](https://user-images.githubusercontent.com/5029719/219416371-f3c462c6-6e83-4914-ac92-055c95aa80cf.gif) ](https://user-images.githubusercontent.com/5029719/219416371-f3c462c6-6e83-4914-ac92-055c95aa80cf.gif) [ ](https://user-images.githubusercontent.com/5029719/219416371-f3c462c6-6e83-4914-ac92-055c95aa80cf.gif)

**Comment by aslng** [comment]
Well. This seems to work with paid/cloud version only! Right?

**Comment by giridharvc7** [comment]
This feature is available for all signed in users in Postman

**Comment by jinyb09017** [comment]
good

**Comment by Ian1971** [comment]
btw I think there is a related bug I found. https://github.com/postmanlabs/postman-app-support/issues/12465

**Comment by Sharjeelkhan914** [comment]
I appreciate @giridharvc7 that this "try" button is added but what if i want to write test cases according to that specific example, how can i write scripts for testing that specific example as of now i can not find a way to do that. Will appreciate more if we can add this feature (using example's request and response body in scripting for test case development) as well.

**Comment by dantebarba** [comment]
+1 in terms of design the examples should have nearly the same attributes than the main request, they should only share the URL and the method. By default they should inherit all the fields (headers, scripts, et al) but they should be editable on each example.

**Comment by dantebarba** [comment]
There is also no option to change the authorization. If I want to create an example that only changes the user I'm logging in I'm unable to do that.

**Comment by bgaudette-ps** [comment]
Once I connect an API to an Azure DevOps repository, the try button disappear.
